src/wallet/scriptpubkeyman.cpp

```diff
@@ -23,6 +23,11 @@ const uint32_t BIP32_HARDENED_KEY_LIMIT = 0x80000000;
 
 bool LegacyScriptPubKeyMan::GetNewDestination(const OutputType type, CTxDestination& dest, std::string& error)
 {
+    if (LEGACY_OUTPUT_TYPES.count(type) == 0) {
```

Comment:
Have you thought about placing asserts in `LegacyScriptPubKeyMan::KeepDestination`, `LegacyScriptPubKeyMan::GetKeyFromPool`, and `LegacyScriptPubKeyMan::LearnRelatedScripts`?

Reply:
Added asserts for all of the `LegacyScriptPubKeyMan` functions which take an `OutputType`.